awx/main/tasks.py

```diff
@@ -915,8 +915,11 @@ def run(self, pk, isolated_host=None, **kwargs):
             credentials = []
             if isinstance(instance, Job):
                 credentials = instance.credentials.all()
+            elif isinstance(instance, InventoryUpdate):
+                # TODO: allow multiple custom creds for inv updates
+                credentials = [instance.get_cloud_credential()]
             elif hasattr(instance, 'credential'):
-                # once other UnifiedJobs (project updates, inventory updates)
+                # once other UnifiedJobs (project updates)
```

Comment:
Does it make sense to just change this now to something more explicit, e.g.,

`elif isinstance(instance, ProjectUpdate)` ?

Reply:
answering this for myself...

```python
In [6]: for cls in UnifiedJobTemplate.__subclasses__():
   ...:     try:
   ...:         cls._meta.get_field('credential')
   ...:         print str(cls) + ' has credential '
   ...:     except:
   ...:         print str(cls) + ' does not have it'
   ...:         
<class 'awx.main.models.projects.Project'> has credential 
<class 'awx.main.models.inventory.InventorySource'> does not have it
<class 'awx.main.models.jobs.JobTemplate'> does not have it
<class 'awx.main.models.jobs.SystemJobTemplate'> does not have it
<class 'awx.main.models.workflow.WorkflowJobTemplate'> does not have it
```

yeah, okay, so that really is down to _just_ Project.

I think we only use this list of `credentials` for injecting custom credentials. I don't know if there's any supported use case for projects, but that still seems sensible in a certain way. Maybe the git or hg modules could use credentials in a way we didn't anticipate.